crates/bevy_pbr/src/light_probe/environment_filter.wgsl

```diff
@@ -0,0 +1,427 @@
+            sample_color = tonemap(sample_color);
+            
+            // Accumulate weighted sample, including geometric term
+            radiance += sample_color * NoL * G;
+            total_weight += NoL * G;
+        }
+    }
+    
+    // Normalize by total weight
+    if (total_weight > 0.0) {
+        radiance = radiance / total_weight;
+    }
+
+    // Reverse tonemap
+    radiance = reverse_tonemap(radiance);
+    
+    // Write result to output texture
+    textureStore(output_texture, coords, face, vec4f(radiance, 1.0));
+}
+
+@compute
+@workgroup_size(8, 8, 1)
+fn generate_irradiance_map(@builtin(global_invocation_id) global_id: vec3u) {
+    let size = textureDimensions(output_texture).xy;
+    let invSize = 1.0 / vec2f(size);
+    
+    let coords = vec2u(global_id.xy);
+    let face = global_id.z;
+    
+    if (any(coords >= size)) {
+        return;
+    }
+    
+    // Convert texture coordinates to direction vector
+    let uv = (vec2f(coords) + 0.5) * invSize;
+    let normal = sample_cube_dir(uv, face);
+    
+    // Create tangent space matrix
+    let tangent_frame = calculate_tangent_frame(normal);
+    
+    // Get blue noise offset for stratification
+    let blue_noise = blue_noise_offset(coords);
+    
+    var irradiance = vec3f(0.0);
+    var total_weight = 0.0;
+    
+    let sample_count = min(constants.sample_count, 64u);
+    
+    for (var i = 0u; i < sample_count; i++) {
+        // Using a predefined set of directions provides good hemisphere coverage for diffuse
+        var dir = get_uniform_direction((i + u32(coords.x * 7u + coords.y * 11u + face * 5u)) % 64u);
+        
+        // Ensure the direction is in the hemisphere defined by the normal
+        let NoL = dot(normal, dir);
+        
+        // Flip the direction if it's in the wrong hemisphere
+        if (NoL < 0.0) {
+            dir = -dir;
+        }
+        
+        // Recalculate NoL after possible flipping
+        let weight = max(dot(normal, dir), 0.0);
+        
+        if (weight > 0.0) {
+            // Lambert PDF
+            let pdf = weight / PI;
+            let width = f32(size.x);
+
+            // Filtered importance sampling
+            let mip_level = clamp(
+                calculate_environment_map_lod(pdf, width, f32(sample_count)),
+                1.0, 
+                constants.roughness * 3.0
+            );
+            
+            // Sample environment with the calculated mip level
+            let sample_color = sample_environment(dir, mip_level).rgb;
+            
+            // Accumulate the sample
+            irradiance += sample_color * weight;
+            total_weight += weight;
+        }
+    }
+    
+    // Normalize and scale by PI for diffuse BRDF
+    if (total_weight > 0.0) {
+        irradiance = irradiance / total_weight * PI;
+    }
+    
+    // Add some low-frequency ambient term to avoid completely dark areas
```

Comment:
Why does this matter?

Reply:
Removed the ambient term, was for testing the texture values in the Metal debugger and in the viewport.